Sources/Testing/ProcedureKitTestCase.swift

```diff
@@ -76,4 +116,38 @@ open class ProcedureKitTestCase: XCTestCase {
             }
         }
     }
+
+    // MARK: Stress Tests
+
+    public func stress(atLevel level: StressLevel = .medium, withName name: String = #function, withTimeout timeoutOverride: TimeInterval? = nil, block: (Int, Int, DispatchGroup) -> Void) {
+        let stressTestName = "Stress Test: \(name)"
+        let timeout: TimeInterval = timeoutOverride ?? level.timeout
+
+        print("\(stressTestName), Parameters: \(level.batches) batches, size \(level.batchSize), timeout: \(timeout)")
+
+        let overallDispatchGroup = DispatchGroup()
+        weak var overallExpectation = expectation(description: stressTestName)
+
+        queue.delegate = nil
+
+        (0..<level.batches).forEach { batch in
+            autoreleasepool {
+                let batchStartTime = CFAbsoluteTimeGetCurrent()
+                (0..<level.batchSize).forEach { iteration in
+                    block(batch, iteration, overallDispatchGroup)
+                }
+
+                let batchFinishTime = CFAbsoluteTimeGetCurrent()
+                let batchDuration = batchFinishTime - batchStartTime
+                print("\(stressTestName), finished batch: \(batch), in \(batchDuration) seconds")
+            }
+        }
+
+        overallDispatchGroup.notify(queue: .main) {
```

Comment:
Yeah, I did have it like this in an previous commit, I'm just going to rethink how it will work. Currently we have one `XCTestCase` subclass, which then has many tests inside it.

I think what I'm going to change is to make a `StressTestCase` class (subclass of `ProcedureKitTestCase`) and then provide APIs on here to perform functions `beforeBatches`, `beforeIterations`, `afterIterations` and `afterBatches`, so that each stress test will be a single instance of `StressTestCase` with a single test in it, which calls the `stress` function.